python/cugraph/dask/pagerank/pagerank.py

```diff
@@ -0,0 +1,253 @@
+       which is 30.
+
+    Returns
+    -------
+    PageRank : dask_cudf.DataFrame
+        Dask GPU DataFrame containing two columns of size V: the vertex
+        identifiers and the corresponding PageRank values.
+
+    Examples
+    --------
+    >>> import dask_cugraph.pagerank as dcg
+    >>> chunksize = dcg.get_chunksize(edge_list.csv)
+    >>> ddf_edge_list = dask_cudf.read_csv(edge_list.csv,
+    >>>                                    chunksize = chunksize,
+    >>>                                    delimiter='\t',
+    >>>                                    names=['src', 'dst'],
+    >>>                                    dtype=['int32', 'int32'])
+    >>> pr = dcg.pagerank(ddf_edge_list, alpha=0.85, max_iter=50)
+    """
+
+    client = default_client()
+    gpu_futures = _get_mg_info(edge_list)
+    npartitions = len(gpu_futures)
+
+    host_dict = _build_host_dict(gpu_futures, client).items()
+    if len(host_dict) > 1:
+        raise Exception("Dask cluster appears to span hosts. Current "
+                        "multi-GPU version is limited to single host")
+
+    master_host = [(host, random.sample(ports, 1)[0])
+                   for host, ports in host_dict][0]
+
+    host, port = master_host
+    gpu_futures_for_host = list(filter(lambda d: d[0][0] == host,
+                                       gpu_futures))
+    exec_node = (host, port)
+    # build ipc handles
+    gpu_data_excl_worker = list(filter(lambda d: d[0] != exec_node,
+                                       gpu_futures_for_host))
+    gpu_data_incl_worker = list(filter(lambda d: d[0] == exec_node,
+                                       gpu_futures_for_host))
+
+    ipc_handles = [client.submit(get_ipc_handle, future, workers=[worker])
+                   for worker, future in gpu_data_excl_worker]
+
+    raw_arrays = [future for worker, future in gpu_data_incl_worker]
+
+    pr = client.submit(_mg_pagerank,
+                       (ipc_handles, raw_arrays, alpha, max_iter),
+                       workers=[exec_node]).result()
+
+    ddf = dc.from_cudf(pr, npartitions=npartitions)
+    return ddf
+
+
+def find_dev(df):
+    gpu_array_src = df['src']._column._data.mem
+    dev = device_of_devicendarray(gpu_array_src)
+    return dev
+
+
+def _get_mg_info(ddf):
+    # Get gpu data pointers of columns of each dataframe partition
+
+    client = default_client()
+
+    parts = ddf
+    key_to_part_dict = dict([(str(part.key), part) for part in parts])
+    who_has = client.who_has(parts)
+    worker_map = []
+    for key, workers in who_has.items():
+        worker = parse_host_port(first(workers))
+        worker_map.append((worker, key_to_part_dict[key]))
+
+    gpu_data = [(worker, client.submit(to_gpu_array, part, workers=[worker]))
+                for worker, part in worker_map]
+
+    wait(gpu_data)
+    [client.submit(find_dev, part, workers=[worker]).result()
+     for worker, part in worker_map]
+
+    return gpu_data
+
+
+def get_n_gpus():
```

Comment:
get_num_gpus or get_number_of_gpus might be clearer. I wasn't sure what this function is just seeing the function name.

Reply:
that is consistent with dask-cudas get_n_gpus function

Reply:
Resolved then. We should be consist with existing dask-* packages